### PR TITLE
jpmsru: Added Lucky Casino

### PR DESCRIPTION
**Pull request by SomeRandomGuyIdk**
- Added a new game, Lucky Casino
- Added 50p slot and lockout to Super 2

### CODE REVIEW

File: src/mame/drivers/jpmsru.cpp [hunk 1]
```diff
@@ -926,6 +1031,55 @@ static INPUT_PORTS_START( j_la )
...
+	PORT_DIPNAME( 0x80, 0x00, "Win Freeze" )
+	PORT_DIPSETTING(    0x00, DEF_STR(Off) )
+	PORT_DIPSETTING(    0x80, DEF_STR(On) )
+
+	PORT_MODIFY("IN2")
+	PORT_DIPNAME( 0x01, 0x00, "Test Hold" ) // Always gives holds if door is open
+	PORT_DIPSETTING(    0x00, DEF_STR(Off) )
+	PORT_DIPSETTING(    0x01, DEF_STR(On) )
+	PORT_BIT( 0x02, IP_ACTIVE_HIGH, IPT_BUTTON1 ) PORT_NAME("Take Win")
+	PORT_BIT( 0x04, IP_ACTIVE_HIGH, IPT_BUTTON2 ) PORT_NAME("Take Half")
```
Comment (cuavas) [inline src/mame/drivers/jpmsru.cpp:1059]: There are `IPT_GAMBLE_TAKE` and `IPT_GAMBLE_HALF` types for these.

File: src/mame/mame.lst [hunk 2]
```diff
@@ -19153,6 +19153,8 @@ j_plus2                         // Plus 2 (CTL)
 j_goldn2                        // Golden 2's (CTL)
 j_sup2p                         // Super 2p Shuffle (MDM)
 j_la                            // Lucky Aces
+j_lc                            // Lucky Casino (JPM)
+j_lca                           //
```
Comment (cuavas) [inline src/mame/mame.lst:19157]: Please sort these by short name within the section for the file.  It makes it easier to find things.

File: src/mame/drivers/jpmsru.cpp (hunk 1 above)
Comment (SomeRandomGuyIdk) [reply src/mame/drivers/jpmsru.cpp]: Fixed.

File: src/mame/mame.lst (hunk 2 above)
Comment (SomeRandomGuyIdk) [reply src/mame/mame.lst]: Fixed.

**Pull request merged**